client/store/pages/dataExport/slice.ts

```diff
@@ -0,0 +1,53 @@
+import { createSlice, Reducer } from '@reduxjs/toolkit'
+
+import { DataExportCountriesAction, DataExportSelectionAction } from './actionTypes'
+// import { AppActions } from '@webapp/store/app'
+// import { HomeActions } from '@webapp/store/page/home'
+import { DataExportState } from './stateType'
+
+const initialState: DataExportState = {
+  countries: [],
+  selection: {
+    countryISOs: [],
+    sections: {},
+  },
+}
+
+export const dataExportSlice = createSlice({
+  name: 'dataExport',
+  initialState,
+  reducers: {
+    reset: () => initialState,
+    updateCountries: (state: DataExportState, action: DataExportCountriesAction) => {
+      state.countries = action.payload
+    },
+
+    updateSelection: (state: DataExportState, { payload }: { payload: DataExportSelectionAction }) => {
+      state.selection = payload.selection
+    },
+  },
+  // extraReducers: (builder) => {
+  //   builder.addCase(AppActions.updateCountryIso, () => initialState)
+  //   builder.addCase(HomeActions.updateCountriesFilter, (state) => {
+  //     state.countries = []
+  //     state.selection = Object.entries(state.selection).reduce<Record<string, DataExportSelection>>(
+  //       (accumulator, [section, sectionSelection]) => {
+  //         return {
+  //           ...accumulator,
+  //           [section]: {
+  //             ...sectionSelection,
+  //             countryISOs: [],
+  //           },
+  //         }
+  //       },
+  //       {}
+  //     )
+  //   })
+  // },
+})
+
+export const DataExportActions = {
+  ...dataExportSlice.actions,
+}
+
+export default dataExportSlice.reducer as Reducer<DataExportState>
```

Comment:
why is this cast needed?

Reply:
It is not needed but it's everywhere. I'm removing it from this file but a small refactor is needed. I'm going to open an issue.